### ISSUE TITLE
ConnectionError

### ISSUE THREAD

**Issue by latkins** [dmjio/hackernews#16]
Not sure what I might be doing wrong, but:

```haskell
main :: IO ()
main = do
  r <- hackerNews (getItem $ ItemId 10575270)
  print r
```

results in:

```
Left ConnectionError
```

Any idea why this might be?

**Comment by zudov** [comment]
Most likely something is actually wrong with your connection.

We throw away all information about the problem when the request fails. https://github.com/dmjio/hackernews/blob/master/src/Web/HackerNews/Client.hs#L133

This means that you would have to figure out what's wrong with connection yourself. To do that you can run something like `withOpenSSL $ openConnectionSSL ctx "hacker-news.firebaseio.com" 443` from ghci and see what exception it throws.

I think we should provide information about what went wrong rather than just giving `ConnectionError`, but it might be not simple with ghcjs (unless we define our own error description)

**Comment by latkins** [comment]
Hmm, so running

```haskell
  ctx <- baselineContextSSL
  x <- withOpenSSL $ openConnectionSSL ctx "hacker-news.firebaseio.com" 443
  print x
```
gives Segmentation fault: 11. This seems to happen before the ```ctx <-
baselineContextSSL```. Guess I will try and work out what is going on!

**Comment by zudov** [comment]
Looks like a problem with some native dependencies

**Comment by latkins** [comment]
Hm, so actually it looks I just had to initialise openssl.

```haskell
main = withOpenSSL $ do
  ctx <- baselineContextSSL
  x <- withOpenSSL $ openConnectionSSL ctx "hacker-news.firebaseio.com" 443
  print x
```
This gives:

```
Host: hacker-news.firebaseio.com
```

Which would suggest something else is happening.

**Comment by latkins** [comment]
Scratch that, the original function is now working. How very odd.

**Comment by zudov** [comment]
> Which would suggest something else is happening.

I guess it means that the connection was established successfully

**Comment by latkins** [comment]
Okay, I understand now. It seems that the failed attempts were all on the Ask sub board. Individual comments, stories etc, all work perfectly. Is this as it should be (e.g. 10027102 fails with getItem) ?

**Comment by dmjio** [comment]
@latkins, yea, if you don't use `withOpenSSL` the `HsOpenSSL` package will segfault :smile:

**Comment by latkins** [comment]
Just to clarify:

```haskell
main :: IO ()
main = do
  a <- hackerNews (getItem $ ItemId 10572982)
  print a

  b <- hackerNews (getItem $ ItemId 10575270)
  print b
```
returns:

```
Left NotFound
Right (ItemStory (Story {storyBy = "kevinchen", storyId = StoryId 10575270, storyKids = [10576385,10575656,10576293,10576021,10575794,10575711,10575512], storyScore = 80, storyTime = 2015-11-16 16:22:24 UTC, storyTitle = "Intel 4004 mask design released, converted to SVG file", storyType = "story", storyUrl = "http://www.4004.com/2015-news.html", storyDeleted = False, storyDead = False}))
```

**Comment by dmjio** [comment]
@latkins cross check `10572982` w/ a different hacker news API. If it's the same issue, it's not on our end.

**Comment by latkins** [comment]
@dmjio:

Using the python firebase library to get 10572982  gives:
```
{u'by': u'sharmi',
 u'descendants': 7,
 u'id': 10572982,
 u'kids': [10573135, 10575020, 10574117, 10573862],
 u'score': 18,
 u'text': u'Hi,\n   I am using Introduction to Probability by Joe Blitzstein and Jessica Hwang to learn probability by self study but I would like to have an additional book for problems practice. Can you suggest one that has solutions available for students?  It would be nice if it is free but it is ok otherwise.',
 u'time': 1447655872,
 u'title': u'Ask HN: What is the best book to learn probability through self study?',
 u'type': u'story'}
```

**Comment by dmjio** [comment]
@latkins this is fixed in `1.0`, on hackage
